OpenSim/Moco/MocoGoal/MocoOrientationTrackingGoal.cpp

```diff
@@ -82,9 +82,21 @@ void MocoOrientationTrackingGoal::initializeOnModelImpl(const Model& model)
         // Check that the reference state names match the model state names.
         checkLabelsMatchModelStates(model, statesTableToUse.getColumnLabels());
 
+        // Check that all coordinates are in the table
+        const CoordinateSet& coords = model.getCoordinateSet();
+        std::vector<std::string> columns = statesTableToUse.getColumnLabels();
+        for (int i = 0; i < coords.getSize(); ++i) {
+            const Coordinate& c = coords.get(i);
+            std::string path = fmt::format("{}/value", c.getAbsolutePathString());
+            OPENSIM_THROW_IF(
+                std::find(columns.begin(), columns.end(), path) == columns.end(),
+                Exception,
+                fmt::format("Coordinate {} not found in states table.", path));
```

Comment:
In exception messages, I like to include what was expected by the user. Consider a message like, "Expected the states reference table to contain columns for all coordinate values in the model, but {} was not found".